src/actors/context-menus/cult-context-menu.ts

```diff
@@ -1,37 +1,24 @@
 import { RqgActorSheet } from "../rqgActorSheet";
 import { RqgActor } from "../rqgActor";
-import { getDomDataset, getGame, getRequiredDomDataset, localize, RqgError } from "../../system/util";
+import { findDatasetValueInSelfOrAncestors, getDomDataset, getGame, getRequiredDomDataset, localize, RqgError } from "../../system/util";
 import { ContextMenuRunes } from "./contextMenuRunes";
 import { RqgItem } from "../../items/rqgItem";
 import { ItemTypeEnum } from "../../data-model/item-data/itemTypes";
+import { Rqid } from "../../system/api/rqidApi";
 
 export const cultMenuOptions = (actor: RqgActor): ContextMenu.Item[] => [
   {
     name: localize("RQG.ContextMenu.ViewDescription"),
     icon: ContextMenuRunes.ViewDescription,
     condition: (el: JQuery) => {
-      const itemId = getDomDataset(el, "item-id");
-
-      let firstItemEl = el[0];
-      while ((firstItemEl?.previousElementSibling as HTMLElement)?.dataset?.itemId === itemId) {
-        firstItemEl = firstItemEl.previousElementSibling as HTMLElement;
-      }
-      return !!firstItemEl.dataset.journalId;
+      const rqid = findDatasetValueInSelfOrAncestors(el[0] as HTMLElement, "cultRqid");
+      return rqid ? true: false;
```

Comment:
I think you could simplify this to `return !!findDatasetValueInSelfOrAncestors(el[0] as HTMLElement, "cultRqid");`

Also in a bunch of more places

Reply:
We will be messing with this again when/if we do the optional uuid based behavior.   Just so we're on the same page, if you drop a Document that doesn't have rqid stuff in the flags it could still link it (perhaps with a warning) and just save the uuid as a property of RqidLink.  In this case we'd have to put the uuid in the dataset if there's no rqid to put there, in which case this code will have to be re-worked to look for the uuid if it doesn't find an rqid.  So the question is, "Do we want this uuid backup behavior?  And if so should I do it now?"

Reply:
Yes I think a uuid "backup" is a good thing (needs some more thought though), and yes we could wait with that. Resolving this comment.